editor/create_dialog.cpp

```diff
@@ -158,12 +158,11 @@ Ref<Texture> CreateDialog::_get_editor_icon(const String &p_type) const {
 	}
 
 	if (ScriptServer::is_global_class(p_type)) {
-		RES icon = ResourceLoader::load(EditorNode::get_editor_data().script_class_get_icon_path(p_type));
-		if (icon.is_valid())
-			return icon;
-		icon = get_icon(ScriptServer::get_global_class_base(p_type), "EditorIcons");
-		if (icon.is_valid())
-			return icon;
+		String icon_path = EditorNode::get_editor_data().script_class_get_icon_path(p_type);
+		if (icon_path.length())
+			return ResourceLoader::load(icon_path);
```

Comment:
Since you no longer do validation of the `load` result, what happens if `icon_path` is a valid string but a wrong path/path to an invalid image?

Reply:
Well, that ideally shouldn't happen because the GDScript parser will generate an error if you supply a path that is invalid. But better safe than sorry considering someone could always just directly edit the project files with an invalid path.

Reply:
Updated it to use `FileAccess::exists()` instead.

Reply:
I was more referring to checking that you get a valid RES with `is_valid()`, as done before. If the file exists but it's actually an OGG, what happens?

Reply:
You know, https://twitter.com/sempf/status/514473420277694465 ;)

Reply:
Lol, I gotcha now. Yeah, very good point. Sorry to burn your CI engines. XD It's updated again now. Should fix both potential problems.

Reply:
Still not good as now there's a branch which does not return :)

Reply:
This would work:
```
		String icon_path = EditorNode::get_editor_data().script_class_get_icon_path(p_type);
		RES icon;
		if (FileAccess::exists(icon_path)) {
			icon = ResourceLoader::load(icon_path);
		}
		if (!icon.is_valid()) {
			icon = get_icon(ScriptServer::get_global_class_base(p_type), "EditorIcons");
		}
		return icon;
```
(provided `RES icon;` does qualify as "invalid" if the first branch is false)

Reply:
Yeah, it would qualify as invalid. That works. :-)